src/d3a/models/strategy/pv.py

```diff
@@ -99,14 +109,15 @@ def produced_energy_forecast_kWh(self):
         # They can be found in the tools folder
         # A fit of a gaussian function to those data results in a formula Energy(time)
         for slot_time in generate_market_slot_list(self.area):
-            difference_to_midnight_in_minutes = \
-                slot_time.diff(self.midnight).in_minutes() % (60 * 24)
-            self.energy_production_forecast_kWh[slot_time] = \
-                self.gaussian_energy_forecast_kWh(
-                    difference_to_midnight_in_minutes) * self.panel_count
-            self.state.available_energy_kWh[slot_time] = \
-                self.energy_production_forecast_kWh[slot_time]
-            assert self.energy_production_forecast_kWh[slot_time] >= 0.0
+            if slot_time >= self.area.now:
```

Comment:
Sorry, i don't get it: Why is this check needed?

Reply:
This is needed now because this method is not only called at the start of the simulation, but can be called in-between via an event. In that case, if this check does not exist, self.state.available_energy_kWh will be modified even for past market slots, and this is not something that we want, since the event is supposed to modify the state of the strategy from this moment and on.

Reply:
Ah yes, thank you!